reportssql/testing_report_functions.sql

```diff
@@ -1,5 +1,37 @@
 -- Testing Report
 
+DROP FUNCTION IF EXISTS Testing_Indicator1;
+
+DELIMITER $$
+CREATE FUNCTION Testing_Indicator1(
+    p_startDate DATE,
+    p_endDate DATE,
+    p_startAge INT(11),
+    p_endAge INT (11),
+    p_includeEndAge TINYINT(1),
+    p_gender VARCHAR(1),
+    p_hivResult VARCHAR(8),
+    p_testingEntryPoint VARCHAR(50)) RETURNS INT(11)
+    DETERMINISTIC
+BEGIN
+    DECLARE result INT(11) DEFAULT 0;
+
+    SELECT
+        COUNT(DISTINCT pat.patient_id) INTO result
+    FROM
+        patient pat
+    WHERE
+        patientGenderIs(pat.patient_id, p_gender) AND
+        getPatientHIVTestDateFromCounselingForm(pat.patient_id) > TIMESTAMPADD(MONTH, -3, CURDATE()) AND
+        getPatientHIVTestDateFromCounselingForm(pat.patient_id) BETWEEN p_startDate AND p_endDate AND
```

Comment:
This should be a different function. There's no date for the current test actually, maybe the test result obs date could be used to check that it was done in the reporting period?